### PR TITLE
Add renaming graphs

### LINKED ISSUE #17: Renaming graphs
It would be a nice addition to have a possibility to rename each graph 😄

### PR DESCRIPTION
**Pull request by v0idzz**
Resolves #17

### CODE REVIEW

**Comment by Ilia-Kosenkov** [comment]
I checked out your brunch and run your code. I see something like this

File: src/dotnet-counters-ui/dotnet-counters-ui/ViewModels/CounterGraphViewModel.cs [hunk 1]
```diff
@@ -24,9 +24,17 @@ public class CounterGraphViewModel : ReactiveObject, IDisposable
     private readonly ObservableCollection<AddedCounterViewModel> _counters = new();
 
     public PlotModel Model { get; }
-    
+
+    public string Name
+    {
+        get => _name;
+        set => this.RaiseAndSetIfChanged(ref _name, value);
+    }
```
Comment (Ilia-Kosenkov) [inline src/dotnet-counters-ui/dotnet-counters-ui/ViewModels/CounterGraphViewModel.cs:32]: I do not see the usage of this property anywhere, nor does VS. Am I missing something?

Comment (v0idzz) [reply src/dotnet-counters-ui/dotnet-counters-ui/ViewModels/CounterGraphViewModel.cs:32]: Here https://github.com/kkokosa/dotnet-counters-ui/pull/38/files#diff-4859752bac924c6e5db65be95a95e6f8459a475dcfc3049fc2f39c0f1f0226d1R33

Comment (Ilia-Kosenkov) [reply src/dotnet-counters-ui/dotnet-counters-ui/ViewModels/CounterGraphViewModel.cs:32]: Nvm, I found the usage. Some weird VS glitch and me not paying attention.

**Comment by v0idzz** [comment]
@Ilia-Kosenkov 
We could surely make it trigger rename on double click. Not sure if that would be more or at least equally as intuitive as a button. If what you are worried about is accidentally pressing 'delete' instead of 'rename', maybe let's add delete confirmation?

**Comment by Ilia-Kosenkov** [comment]
Ok, what do you think of moving the 'Delete' button somewhere to the corner? 
image

It is of course Windows-style (top right corner), but it is at least separated from renaming. And leave the rename button right next to the name itself, as it is now.

As a minor addition, could we have the textbox automatically focused and text inside selected when clicking 'Edit' button? Right now you click the button, you start typing and realize you are typing to `/dev/null`, you need an extra click to get the proper field focused.

**Comment by kkokosa** [comment]
I like @Ilia-Kosenkov proposal to make delete button top right and rename in the middle. 

BTW adding confirmation to delete a whole graph may be a nice feature indeed, I'll add a new issue :)

**Comment by v0idzz** [comment]
All of the issues above have been addressed in the recent commits

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs [hunk 2]
```diff
@@ -30,13 +32,63 @@ public partial class GraphToolbox : UserControl
...
+    {
+        SetRenameMode(false);
+    }
+
+    private void SetRenameMode(bool renameMode)
+    {
+        if (renameMode)
+        {
+            _renameBox.Focus();
+            _renameBox.CaretIndex = _renameBox.Text.Length;
```
Comment (Ilia-Kosenkov) [inline src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs:88]:
```suggestion
            _renameBox.Focus();
            _renameBox.SelectAll();
            _renameBox.CaretIndex = _renameBox.Text.Length;
```

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml [hunk 3]
```diff
@@ -5,15 +5,20 @@
...
-                    CommandParameter="{Binding #Ctrl.DeleteCommandParameter}">
-                <PathIcon Data="{StaticResource delete_regular}" />
-            </Button>
-        </StackPanel>
+             x:Name="Ctrl"
+             Padding="30, 0">
+    <Grid ColumnDefinitions="25,*,Auto,*,25">
+        <TextBox Name="RenameBox" Grid.Column="2" Text="{Binding #Ctrl.Text}" IsVisible="False"
+                 KeyUp="RenameBox_OnKeyUp" LostFocus="RenameBox_OnLostFocus" />
+        <TextBlock Name="NameBlock" Grid.Column="2" Text="{Binding #Ctrl.Text}" VerticalAlignment="Center" />
```
Comment (Ilia-Kosenkov) [inline src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml:13]: For some reason, `TextBox` no longer shrinks correctly. If you rename your graph to `"A"`, the editable `TextBox` will be quite large, but when the editing is finished, the `TextBlock` shrinks to one-letter size. Maybe there can be a minimum width for both controls to synch their behavior?

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs [hunk 4]
```diff
@@ -30,13 +32,63 @@ public partial class GraphToolbox : UserControl
...
+        SetRenameMode(true);
+    }
+
+    private void RenameBox_OnKeyUp(object? sender, KeyEventArgs e)
+    {
+        if (e.Key == Key.Enter)
+        {
+           SetRenameMode(false);
+        }
+    }
```
Comment (Ilia-Kosenkov) [inline src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs:76]: Idea for a future PR -- on `Escape` cancel changes and keep the original name of the graph.

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs (hunk 2 above)
Comment (Ilia-Kosenkov) [reply src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs:88]: Again, modeling the most common behavior for renaming things -- you hit rename, you get the previous name selected for you, and caret/cursor put at the end of the name.

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml (hunk 3 above)
Comment (Ilia-Kosenkov) [reply src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml]: Looks better now, thanks!

**Review by Ilia-Kosenkov — APPROVED** [review]
LGTM, thanks!

File: src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs (hunk 4 above)
Comment (v0idzz) [reply src/dotnet-counters-ui/dotnet-counters-ui/Views/GraphToolbox.axaml.cs]: Also done :)

**Comment by Ilia-Kosenkov** [comment]
Discarding the new graph name works perfectly!

**Pull request merged**